include/revng/Model/DisassemblyConfiguration.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+/* TUPLE-TREE-YAML
+name: DisassemblyConfiguration
+type: struct
+fields:
+  - name: RawByteColumnWidth
+    doc: |
+      Set this to 0 (default) to disable the raw byte emission.
+      Set it to 1 for non-column view (since each instruction is guaranteed
+      to have at least one byte).
+    type: uint64_t
+    optional: true
+  - name: InstructionAddressColumnWidth
+    doc: |
+      Set this to 0 (default) to disable the instruction address emission.
+      Set it to 1 for non-column view (since each instruction is guaranteed
+      to have an address).
```

Comment:
I'm not sure what "non-column view" means and what happens if it's greater than `1`. Do we wrap the address?

Reply:
`non-column view` means `no tabulation`, as in, the instruction just starts after a space.

Reply:
Wait, are we emitting `\t`? Those are evil since you have somehow to assume the tabulation size.
If not, then please make a specific example of each option because it's unclear to me.

Reply:
:_t_burn:!!

We imitate it with spaces.

> example

So, if `RawByteColumnWidth` is set to `10`, then we would get something like:
```cpp
01        one_byte_instrunction
01 02     two_byte_instrunction
01 02 03 04 four_byte_instrunction
01 02 03 04 05 06 07 08 eight_byte_instrunction
```

Setting it to `20` leads to
```cpp
01                 one_byte_instrunction
01 02              two_byte_instrunction
01 02 03 04        four_byte_instrunction
01 02 03 04 05 06 07 08 eight_byte_instrunction
```

And so on.

Reply:
The behave exactly the same.

Reply:
Can we have no configuration and simply right align addresses? Handling 32-bit and 64-bits as appropriate

32-bits:
```
    0x1000 lol
0x10001000 lel
```
64 bits:

```
        0x10001000 lol
0x1000100010001000 lel
```

Alternatively we can precompute the maximum size of an address in the current function.

Left aligning numbers is very sad.

Reply:
Done.